api/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -62,6 +62,17 @@ func (r *OpenTelemetryCollector) ValidateCreate() error {
 	if r.Spec.Mode != ModeStatefulSet && len(r.Spec.VolumeClaimTemplates) > 0 {
 		return errors.New("Can only specify VolumeClaimTemplates for statefulsets")
 	}
+
+	// validate replicas for sidecar and daemonsets mode
```

Comment:
I think it might be worth extracting the common validation checks to a specific function. Right now, ValidateCreate and ValidateUpdate are pretty much the same.